spec/reek/smells/too_many_constants_spec.rb

```diff
@@ -0,0 +1,125 @@
+require_relative '../../spec_helper'
+require_lib 'reek/smells/too_many_constants'
+require_relative 'smell_detector_shared'
+
+RSpec.describe Reek::Smells::TooManyConstants do
+  let(:detector) { described_class.new(max_constants: default_max_consts) }
+
+  it_should_behave_like 'SmellDetector'
+
+  def default_max_consts
+    described_class::DEFAULT_MAX_CONSTS
+  end
+
+  def too_many_consts
+    default_max_consts + 1
+  end
+
+  def casgn_sequence(count: default_max_consts, consts: ('A'..'Z').to_a)
+    consts.first(count).map.with_index { |name, i| "#{name}=#{i}" }.join("\n")
+  end
+
+  context 'counting constants' do
+    it 'should not report for non-excessive constants' do
+      src = <<-EOS
+        class Dummy
+          #{casgn_sequence}
```

Comment:
Having this indirection makes the test less readable I think, since it is not immediately clear how many constants are created here. Just writing it all out makes the tests a bit longer but a lot clearer.